packages/package_info/lib/package_info.dart

```diff
@@ -4,19 +4,59 @@
 
 import 'dart:async';
 
+import 'package:meta/meta.dart';
 import 'package:flutter/services.dart';
 
-/// Provides information about the current application package.
 const MethodChannel _kChannel =
     const MethodChannel('plugins.flutter.io/package_info');
 
-/// Returns the `CFBundleShortVersionString` on iOS or `versionName` on Android
-Future<String> get version async => await _kChannel.invokeMethod('getVersion');
+/// Wrap application Bundle (on iOS) and Package (on Android) information,
+/// providing cached verion, build number and package name.
+///
+/// PackgeInfo implemented in singleton, after first called
+/// [PackageInfo.getInstance], cached value never changed.
+///
+/// ```dart
+///     PackageInfo paakcageInfo = await PackageInfo.getInstance()
+///     print("Version is: ${packcageInfo.version}");
+/// ```
+///
+class PackageInfo {
+  PackageInfo._(
+      {@required this.version,
+      @required this.buildNumber,
+      @required this.packageName})
```

Comment:
A trailing comma should hopefully make `dartfmt` produce something prettier here.